api/api/src/main/scala/uk/ac/wellcome/platform/api/models/AggregationSet.scala

```diff
@@ -0,0 +1,87 @@
+package uk.ac.wellcome.platform.api.models
+
+import io.circe.{Decoder, Json}
+import uk.ac.wellcome.models.work.internal.WorkType
+import uk.ac.wellcome.json.JsonUtil._
+
+import scala.util.{Failure, Success}
+
+// TODO: Return a `ParseError`s where applicable
+case class Genre(label: String)
+case class AggregationSet(workType: Option[Aggregation[WorkType]],
+                          genre: Option[Aggregation[Genre]])
```

Comment:
What do you think about removing the `Option[_]` here and just have `Aggregation[_]`? It seems to just contain a list of buckets so could have an empty list when none exist, or does the front-end need to receive `null` in that case?

Reply:
Ah, I just re-read this - we need the option as we can get a request of:
`?aggreagtions=workType` | `?aggreagtions=genre` | `?aggreagtions=workType,genre` and we only attach the aggregations you've requested.

Having an empty list might imply that you've requested it, there just aren't any buckets e.g. none of your results have any genres, or there are no results.

So:

No key on object = client hasn't requested it.
Key with no results = client has requested it, it is just empty.